asllib/Typing.ml

```diff
@@ -2986,24 +3097,28 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
       | Some t1, None -> t1
       | None, None -> Error.fatal_from loc UnrespectedParserInvariant
     in
+    let initial_value'' =
```

Comment:
I think omputing `initial_value'` and then `initial_value''` is very fussy. Why not first compute just `initial_value_type`, then `declared_t` and then `initial_value'`?